Cue/app/tabs/library/CardListView.js

```diff
@@ -10,50 +11,107 @@ import EmptyView from '../../common/EmptyView'
 import CardListViewRow from './CardListViewRow'
 import ListViewHairlineSeparator from '../../common/ListViewHairlineSeparator'
 
+const styles = {
+  liftedRowStyle: {
+    opacity: 1,
+    transform: [{scale: 1.02}],
+    backgroundColor: 'white',
+    overflow: 'visible',
+
+    // Android
+    elevation: 8,
+
+    // iOS
+    shadowColor: 'black',
+    shadowOpacity: 0.35,
+    shadowOffset: {
+      width: 0,
+      height: 4,
+    },
+    shadowRadius: 6,
+  },
+}
+
 type Props = {
   cards: Array<Card>,
-  isFiltering: boolean,
+  editing?: boolean,
+  filtering?: boolean,
   accession?: string,
+  onDeleteCard?: (card: Card) => void,
+  onEditCard?: (card: Card) => void,
   onFlagCard?: (cardUuid: string, flag: boolean) => any,
+  onMoveCard?: (cardUuid: string, from: number, to: number) => any,
 }
 
+// Store the most recent edit mode so that when the edit mode changes,
+// we can force the entire list to re-render without unmounting and remounting
+// which would cause us to lose our scroll position.
+type AnnotatedCard = Card & {_editing: boolean}
+
 export default class CardListView extends React.Component {
   props: Props
 
   state: {
-    dataSource: ListView.DataSource
+    data: {[key: string]: AnnotatedCard}, // maps from card.uuid to card
+    order: Array<string>, // list of card.uuid to maintain ordering
   }
 
   constructor(props: Props) {
     super(props)
 
     this.state = {
-      dataSource: this._getDataSource(this.props.cards, this.props.isFiltering)
+      ...this._getDataAndOrder(this.props.cards, this.props.filtering),
     }
   }
 
   componentWillReceiveProps(newProps: Props) {
     this.setState({
-      ...this.state,
-      dataSource: this._getDataSource(newProps.cards, newProps.isFiltering)
+      ...this._getDataAndOrder(newProps.cards, newProps.filtering)
     })
   }
 
-  _getDataSource = (cards: Array<Card>, isFiltering: boolean) => {
-    let ds = new ListView.DataSource({
-      rowHasChanged: (r1: Card, r2: Card) => {
-        return (r1.front !== r2.front || r1.back !== r2.back || r1.uuid !== r2.uuid ||
-                r1.needs_review !== r2.needs_review)
-      }
-    })
-
+  _getDataAndOrder = (cards: Array<Card>, filtering: boolean) => {
     cards = cards || []
 
-    let rows = isFiltering
+    let rows = filtering
       ? cards.filter((card: Card) => card.needs_review )
       : cards
 
-    return ds.cloneWithRows(rows)
+    let data = rows.reduce((acc: {[key: string]: AnnotatedCard}, row: Card) => {
+      acc[row.uuid] = {...row, _editing: this.props.editing}
+      return acc
+    }, {})
+    let order = rows.map((card: Card) => card.uuid)
+
+    return {
+      data,
+      order,
+    }
+  }
+
+  _rowHasChanged = (r1: AnnotatedCard, r2: AnnotatedCard) => {
+    return r1._editing !== r2._editing
+      || r1.uuid !== r2.uuid
+      || r1.front !== r2.front
+      || r1.back !== r2.back
+      || r1.needs_review !== r2.needs_review
```

Comment:
I think r1 != r2 is enough here? We never edit a card without creating a completely new object